### PR TITLE
[Serialization] Add flag to force unsafe recovery from some xref failures

### PR DESCRIPTION
**Pull request by xymus**
Intro a deserialization mode controlled by the flag `-experimental-force-workaround-broken-modules` to attempt unsafe recovery from deserialization failures caused by project issues.

The one issue handled at this time is when a type moves from one module to another. With this new mode the compiler may be able to pick a matching type in a different module. This is risky to use, but may help in a pinch for a client to fix and issue in a library over which they have no control.

### CODE REVIEW

**Comment by xymus** [comment]
@swift-ci Please smoke test

File: lib/Serialization/Deserialization.cpp [hunk 1]
```diff
@@ -1945,13 +1946,32 @@ ModuleFile::resolveCrossReference(ModuleID MID, uint32_t pathLen) {
...
+        errorKind == ModularizationError::Kind::DeclMoved &&
+        !values.empty()) {
+      // Print the error as a remark and notify of the recovery attempt.
+      llvm::handleAllErrors(std::move(error),
+        [&](const ModularizationError &modularError) {
+          modularError.diagnose(this, DiagnosticBehavior::Remark);
+        });
+      getContext().Diags.diagnose(getSourceLoc(),
+                                  diag::modularization_issue_worked_around,
+                                  foundIn);
```
Comment (bnbarham) [inline lib/Serialization/Deserialization.cpp:1971]: Would it be possible to make this a warning with the remark instead attached as a note?

Comment (xymus) [reply lib/Serialization/Deserialization.cpp]: Good idea. I inverted both diagnostics, using the warning as the main message with the current remark as a note.

**Comment by xymus** [comment]
@swift-ci Please smoke test

**Comment by xymus** [comment]
@swift-ci Please smoke test

**Review by bnbarham — APPROVED** [review]

**Pull request merged**